### ISSUE TITLE
All typesrcipt file are not resolved.

### ISSUE THREAD

**Issue by vipcxj** [eslint/eslint#13363]
**Tell us about your environment**

* **ESLint Version:** 6.8.0 (I can't use 7 because airbnb not support it)
* **Node Version:** 13.6.0
* **npm Version:** 6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// This is generated by the eslint init tool. I have not change anything.
module.exports = {
  env: {
    browser: true,
    es6: true,
    node: true,
  },
  extends: [
    'plugin:react/recommended',
    'airbnb',
  ],
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 11,
    sourceType: 'module',
  },
  plugins: [
    'react',
    '@typescript-eslint',
  ],
  rules: {
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './rootReducer';
```
eslint said
```
ESLint: Missing file extension for "./rootReducer"(import/extensions)
ESLint: Unable to resolve path to module './rootReducer'.(import/no-unresolved)
```
rootReducer.ts exist. without eslint, everything work well.
with eslint, all ts/tsx files under src are not resolved and eslint require me to add the extension to it. however, this is not permit by ts compiler.

And there is another problem.
`App.tsx`
```ts
import * as React from 'react';

const App = () => {};

export default App;
```
eslint complain that 
```
ESLint: 'React' is defined but never used.(no-unused-vars)
```
But it seems that all tsx files should import `React` module.

**What did you expect to happen?**
eslint should make life happy not make it worse.

**What actually happened? Please include the actual, raw output from ESLint.**

**Are you willing to submit a pull request to fix this bug?**

**Comment by mdjermanovic** [comment]
Hi @vipcxj, thanks for the issue!

```
ESLint: Missing file extension for "./rootReducer"(import/extensions)
ESLint: Unable to resolve path to module './rootReducer'.(import/no-unresolved)
```

These errors are coming from third-party rules and are not related to eslint itself. Can you please open an issue in [benmosher/eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) repository?

```
ESLint: 'React' is defined but never used.(no-unused-vars)
```

This seems like a correct error, there is no need to import from `'react'` if you're not using it in that module?

**Comment by vipcxj** [comment]
> Hi @vipcxj, thanks for the issue!
> 
> ```
> ESLint: Missing file extension for "./rootReducer"(import/extensions)
> ESLint: Unable to resolve path to module './rootReducer'.(import/no-unresolved)
> ```
> 
> These errors are coming from third-party rules and are not related to eslint itself. Can you please open an issue in [benmosher/eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) repository?
> 
> ```
> ESLint: 'React' is defined but never used.(no-unused-vars)
> ```
> 
> This seems like a correct error, there is no need to import from `'react'` if you're not using it in that module?

The App is a function component. It will use the `React` module.

**Comment by mdjermanovic** [comment]
`no-unused-vars` warns about a possible error if you forget to use declared/imported variable. When you use `React`, the warning will disappear.

Rules are configurable, so if you don't want to be notified about unused variables, you can turn this rule off in the config file:

```json
{
  "rules": {
     "no-unused-vars": "off"
  }
}
```

**Comment by vipcxj** [comment]
I think this is a bug about eslint config generate tool. I told the tool I need use typescript, react, and arnbib. but the generate config not work well. I found that arnbib config is designed to work with pure js, not typescript, I need a lot of extra config to make it work with typescript. Then I found [eslint-config-airbnb-typescript](https://www.npmjs.com/package/eslint-config-airbnb-typescript) which has the same rule of arnbib and not complict with react and typescript. I think the generate tool should replace `eslint-config-airbnb with ` with `eslint-config-airbnb-typescript` if the user need typescript support.
